o1-labs/o1js-bindings
# Merge berkeley -> develop

- Fixup bindings for hex serialization
- If recompiling the WASM changes it, something has gone very wrong
- bindings
- separate type checking from build script
- fixup
- move bindings location and add guarding d.ts
- Fixup to make npm run bindings happy
- tweak how merge conflict is resolved
- Update bindings for evaluated selectors
- Update bindings for optional column evaluations
- Run prettier on bindings.js
- wasm/{wasm_vector, plonk_proof?: move WasmVecVecF{p,q}
- wasm/src/index: define wasm lookup table
- wasm/src/index: define wasm runtime table cfg
- Add lookup selector evaluations to proof
- Update bindings
- update bindings
- wasm/src/plonk_proof.rs: define wasm runtime tables
- kimchi/js/bindings.js: add runtime_tables to proof_create
- bindings.js: remove duplicated definition
- kimchi/js/bindings.js: define js/wasm converters for lookup tables
- bindings_js_test: add empty runtime table cfg arg
- wasm/index: add runtime table cfgs arg
- bindings.js: add runtime_table_cfgs arg
- kimchi/js/bindings.js: define converters for runtime table cfg
- Run npm run bindings
- Revert "Merge pull request #86 from o1-labs/fix/premature-vk-hash"
- Expose overrideWrapDomain
- Update bindings
- Update bindings
- Update bindings
- Update bindings
- deduplicate logic of parsing override_wrap_domain
- deduplicate logic of parsing override_wrap_domain
- update bindings
- Run prettier -w kimchi/js/bindings.js
- CI: add basic workflow to run prettier
- Run prettier -w kimchi/js-bindings.js
- CI: add basic workflow to run prettier
- Run prettier -w kimchi/js/bindings.js
- Import prettier config from snarkyjs
- Import prettier config from snarkyjs
- Run prettier -w kimchi/js/bindings.js
- update worker spec
- kimchi/js/bindings.js: fix runtime table cfg field access
- kimchi/js/bindings.js: change converters for runtime, lookup table and runtime table cfg
- kimchi/js/bindings.js: unify field access of caml record
- Run npm run bindings
- Remove lookup tables converters
- Remove runtime table converters
- Remove caml_is_empty_array_caml_array and caml_create_rust_empty_vector
- Regenerate bindings with npm run bindings
- Run npm run bindings
- Add lookup tables in worker-spec + comments
- Add comments related to changes that must appear in worker-spec
- Reintroduce converters for lookup tables and add it in stubs
- binding_js_test.ml: add empty lookup tables
- index: add param lookup tables
- bindings.js: add bytecode version of the caml stubs
- Artefacts: run npm run bindings
- Add runtime tables to worker-spec and add comments
- plonk_proof.rs: pass runtime tables to create proof
- kimchi/js/bindings.js: add runtime_tables to proof_create
- bindings_js_test: add runtime tables
- snarky_bindings: pass empty runtime tables atm
- Reintroduce JS converters for runtime tables
- Artefacts: generate using npm run bindings
- Add js/**/*.js in prettier check
- Run the following command:
- CI: run prettier check on all JS files in kimchi/js
- Replace joo_global_object by globalThis
- Regenerate bindings with npm run bindings
- fix snarky_js_constants definitions
- build bindings
- prettier
- Introduce different Wasm equivalent types and add lookup verif index
- bindings.js: add to/of_rust conversions of new structures
- Update artefacts after run npm run bindings
- Expose lookup commitments
- bindings.js: removing unused parameter
- bindings.js: remove TODO
- bindings.js: use caml_opt_to_rust/of_rust
- Update bindings after artefacts generation
- Introduce different Wasm equivalent types and add lookup verif index
- Update local ledger implementation to match ledger intf
- bindings.js: add to/of_rust conversions of new structures
- Run npm run bindings
- Removing mk_affine in caml_plonk_verification_evals_to_rust
- Removing poly_comm_class from caml_plonk_verification_evals_to_rust
- Ignore warning about unused value
- update bindings
- address feedback - add documentation in comments
- Expose lookup commitments
- bindings.js: removing unused parameter
- Update bindings after artefacts generated
- Remove runtime tables parameter
- Run npm run bindings to generate bindings artefacts
- Removing runtime table parameters in binding_js_test.ml
- update bindings
- Update bindings_js_test with new optional commitments
- Rerun npm run bindings
